src/super_gradients/training/pipelines/pipelines.py

```diff
@@ -68,10 +68,13 @@ def __init__(
         fuse_model: bool = True,
         dtype: Optional[torch.dtype] = None,
     ):
-        model_device: str = infer_model_device(model=model).type
-        self.model = model.to(device) if device and device != model_device else model
-        self.device = device or model_device
+        model_device: torch.device = infer_model_device(model=model)
+        if device:
+            device: torch.device = resolve_torch_device(device=device)
```

Comment:
I thought it would be more clear to add typing `torch.device` because the type is different to the input type `str`